### ISSUE TITLE
Every hour previously justified issues become unjustified for two iam users

### ISSUE THREAD

**Issue by cswilliams** [Netflix/security_monkey#946]
I noticed that every hour, all of the issues for two iam users in my account show back up under the "Aging Issues" dashboard.  When I click in on the items, all of the past justifications that I had added for the issues are gone.  I am able to add justifications again and then the issues will disappear off the "Aging Issues" dashboard until the next hour whereby my justifications will again be cleared out.   For some reason, this only seems to affect two iam users in my account, other resources seem to be fine.

This is using the code on the develop branch.

**Comment by mikegrima** [comment]
Other IAM users with similar circumstances don't have this problem -- only these two?

**Comment by cswilliams** [comment]
Yeah, just these two users (other iam users are fine).  Both of these users only have two issues. One is "Found issue(s) in attached Managed Policy" and the other is "Informational: Active Key".  I justify each of these issues for both iam users and in the next hour when the scheduler runs, they lose these justifications.

I can't really tell what is special about these users.  They both use a managed policy (but not the same policy).  The policies that they relate to also have all their issues justified and are not losing their justifications (it appears just to affect the iam user). The policies also are not particularly complex, just mostly granting S3 access.

I have other iam users that are almost setup identically (use a similar managed policy) and are not affected by this bug, so it is definitely strange.

**Comment by mikegrima** [comment]
Hello @cswilliams Are you still experiencing this issue?

**Comment by artem-smotrakov** [comment]
Hello,

I see the same issue but with multiple IAM roles. I use 1.1.3

I don't have a scheduler, the justifications are gone after I run `monkey audit_changes `

**Comment by artem-smotrakov** [comment]
In my case, I noticed that justified issues become unjustified only if an attached managed policy has an issues. I justified the issues in the attached managed policy but after running `monkey audit_changes` the issue in the role became unjustified again. I am wondering if the problem is in `library_check_attached_managed_policies` in `security_monkey/auditors/iam/iam_policy.py`:

https://github.com/Netflix/security_monkey/blob/b3ef821d0e9070569c4d70a7a6a234393de3750a/security_monkey/auditors/iam/iam_policy.py#L207

It check for issues in attached policy

```
for mp_item in mp_items or [] and not found:
                mp_arn = mp_item.config.get('arn', mp_item.config.get('Arn'))
                if mp_arn == item_mp_arn:
                    found = True
                    if mp_item.db_item.issues:
                        self.link_to_support_item_issues(iam_item, mp_item.db_item, None, "Found issue(s) in attached Managed Policy")
```

But it doesn't check if an issue in an attached policy are justified or not. Should it be something like the following?

```
diff --git a/security_monkey/auditors/iam/iam_policy.py b/security_monkey/auditors/iam/iam_policy.py
index 895cf2e..a8173fd 100644
--- a/security_monkey/auditors/iam/iam_policy.py
+++ b/security_monkey/auditors/iam/iam_policy.py
@@ -207,8 +207,11 @@ class IAMPolicyAuditor(Auditor):
             for mp_item in mp_items or [] and not found:
                 mp_arn = mp_item.config.get('arn', mp_item.config.get('Arn'))
                 if mp_arn == item_mp_arn:
-                    found = True
-                    if mp_item.db_item.issues:
+                    for issue in mp_item.db_item.issues:
+                        if issue.justified:
+                            found = True
+                            break
+                    if found:
                         self.link_to_support_item_issues(iam_item, mp_item.db_item, None, "Found issue(s) in attached Managed Policy")
 
             if not found:
```

**Comment by artem-smotrakov** [comment]
The patch above seems to fix the problem I described although it generates many `IAM Managed Policy defined but not found ...` errors. I'll try to get rid of those errors and create a pull request.
